app/views/buyers/accounts/_account_details.html.slim

```diff
@@ -54,13 +54,13 @@ table.list
                           reject_admin_buyers_account_path(account),
                           method: :post,
                           class: 'reject action'
-      - unless account.suspended_or_scheduled_for_deletion?
+      - unless account.suspended_or_scheduled_for_deletion? || cannot?(:suspend, account)
         = action_link_to 'Suspend',
                           suspend_admin_buyers_account_path(account),
                           method: :post,
                           data: { confirm: 'Are you sure?', disable_with: 'suspending…' },
                           class: 'action suspend'
-      - if can?(:resume, account) && account.can_resume?
+      - if can?(:resume, account) && account.suspended?
```

Comment:
```suggestion
      - if account.suspended? && can?(:resume, account)
```

Just for the consistency with the above line :grimacing: 

It could be either way - first `can?` then state checking, or otherwise, but I believe checking the account state is "cheaper" than checking the ability, so I think it can go first.